### PR TITLE
work around multiple-monitors setup

### PR DESCRIPTION
**Pull request by muphil**

### CODE REVIEW

File: wrapper/xim/xmisc.c [hunk 1]
```diff
@@ -93,10 +93,15 @@ in_range(int x, int min, int max)
...
 {
-    int screen_width, screen_height;
-    get_screen_size(&screen_width, &screen_height);
-    *x = in_range(*x, 0, screen_width - width);
-    *y = in_range(*y, 0, screen_height - height);
+    GdkScreen *screen; 
+    gint mon;
+    GdkRectangle srect;
+    screen = gdk_screen_get_default();
+    mon = gdk_screen_get_monitor_at_point(screen, *x, *y);
```
Comment (tchaikov) [inline wrapper/xim/xmisc.c]: @muphil is it possible to port this change to xlib ?

**Comment by muphil** [comment]
@tchaikov Sorry, I don't quit understand what you mean. I'm not familiar with xlib, but I
think the root of this bug is not related to xlib, gtk enumerates monitors with
xrandr internally. The bug presents when there are "holes" in the virtual
screen, this happens when having more than one monitors which are not arranged
regularly or not of equal size. So instead of adjusting the position of the
preedit window according to the virtual screen, we need to do that according to
the monitor where the preedit window currently is.

**Comment by tchaikov** [comment]
> but I think the root of this bug is not related to xlib, gtk enumerates monitors with
xrandr internally.

@muphil agreed. i was wrong assuming that the xim frontend solely depends on xlib, but it turns out that it uses gtk extensively when it comes to UI drawing. so yeah, let's merge it!

thanks! and happy lunar new year!

**Comment by muphil** [comment]
@tchaikov Thank you, and the same to you!

**Pull request merged**